File: app/src/main/java/com/example/pam_project/utils/TaskStatus.java
package com.example.pam_project.utils;

public enum TaskStatus {
    DONE,
    PENDING;

    public static final String STRING_DONE = "done";
    public static final String STRING_PENDING = "pending";

    public static TaskStatus getStatus(String status) {
        switch (status){
            case "done": return TaskStatus.DONE;
            case "pending": return TaskStatus.PENDING;
            default: return null;
        }
    }

    public static String statusToString(TaskStatus status){
        switch (status){
            case DONE: return STRING_DONE;
            case PENDING: return STRING_PENDING;
            default: return null;
        }
    }
}

File: app/src/main/java/com/example/pam_project/utils/constants/TaskStatus.kt
package com.example.pam_project.utils.constants

enum class TaskStatus {
    DONE, PENDING;

    companion object {
        private const val STRING_DONE = "done"
        private const val STRING_PENDING = "pending"

        fun getStatus(status: String?): TaskStatus? {
            return when (status) {
                "done" -> DONE
                "pending" -> PENDING
                else -> null
            }
        }

        fun statusToString(status: TaskStatus?): String? {
            return when (status) {
                DONE -> STRING_DONE
                PENDING -> STRING_PENDING
                else -> null
            }
        }
    }
}